pkg/router/controller/router_controller.go

```diff
@@ -72,64 +66,158 @@ func (c *RouterController) Run() {
 		c.HandleNamespaces()
 		go utilwait.Forever(c.HandleNamespaces, c.NamespaceSyncInterval)
 	}
-	go utilwait.Forever(c.HandleRoute, 0)
-	go utilwait.Forever(c.HandleEndpoints, 0)
+
+	c.StartRouteWatch(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
```

Comment:
Look at RegisterSharedInformerEventHandlers() in pkg/sdn/plugin/common.go
We could do similar stuff here and that will reduce code duplication for all the router watch resources.